src/main/java/com/google/api/gax/core/PageAccessor.java

```diff
@@ -0,0 +1,21 @@
+package com.google.api.gax.core;
+
+/**
+ * Accessor for paged results from a list API method"
+ */
+public interface PageAccessor<T> extends Iterable<T> {
```

Comment:
I think you should make it clear that `iterator` would return an Iterator with the complete listing result and if
necessary will perform more rpc calls to fetch more pages.

You can do that in the class javadoc description and/or method level (`@Override Iterator<T> iterator()`).